epoxy-sample/src/main/java/com/airbnb/epoxy/sample/models/ColorModel.java

```diff
@@ -54,6 +61,31 @@ public void bindChanges(@NonNull ColorHolder holder,
     }
   }
 
+  @Nullable
+  @Override
+  public EpoxyItemAnimation bindChangesAnimated(@NonNull ColorHolder view,
+                                                @NonNull EpoxyModel<?> previouslyBoundModel,
+                                                @Nullable EpoxyItemAnimation interruptedAnimation) {
+
+    int currentColor = ((ColorModel) previouslyBoundModel).color;
+    float currentRotation = 0f;
+
+    if(interruptedAnimation != null) {
+      AnimatorSet previousAnim = (AnimatorSet) ((EpoxyItemAnimation.AnimatorWrapper) interruptedAnimation).wrappedAnimator;
```

Comment:
Yeah, the casting is a shame. The only thing I can think of is using an interface to specify the model and animation type, and that wouldn't allow people to have two different animation types if they wanted.

Can we have the type just be `Animator` instead of `EpoxyItemAnimation`? If we need EpoxyItemAnimation we could wrap it internally after they pass back an `Animator`. That way they can avoid the `EpoxyItemAnimation.fromAnimator(combinedAnim)` and `((EpoxyItemAnimation.AnimatorWrapper) interruptedAnimation).wrappedAnimator` boilerplate at least.

Is it necessary to get the current animation values off of the previous animation? I would have thought that the view would already be updated with that value, and by starting an animation to a new value it would automatically transition from the previous animation.

Using something like https://github.com/florent37/ViewAnimator for a fluent animation description here would be really nice, and we could encourage usage like that.

With something like that this whole method could be replaced with just
```
return holder.cardView.animate()
        .backgroundColor(color)
        .rotationXBy(360f);
```

The problem with that library, and with the built in android `view.animate()` is that they handle the animations themselves instead of returning an animator object. It would be nice to have a fluent animator language that returns the animator, which would make implementing animations here a breeze

I guess this begs the question, if you can get the current animation values off the view, why do you need the previous animator? I'm not sure...